File: PholioVisualisationWS/Export/SignificanceFormatter.cs
using PholioVisualisation.PholioObjects;

namespace PholioVisualisation.Export
{
    public class SignificanceFormatter
    {
        private readonly int _polarityId;
        private readonly int _comparatorMethodId;

        public SignificanceFormatter(int polarityId, int comparatorMethodId)
        {
            _polarityId = polarityId;
            _comparatorMethodId = comparatorMethodId;
        }

        public string GetLabel(Significance significance)
        {
            // Comparator method first because it overrides polarity if quintiles
            if (_comparatorMethodId == ComparatorMethodIds.Quintiles)
            {
                switch ((int)significance)
                {
                    case 1:
                        return "Lowest quintile";
                    case 2:
                        return "2nd lowest quintile";
                    case 3:
                        return "Middle quintile";
                    case 4:
                        return "2nd highest quintile";
                    case 5:
                        return "Highest quintile";
                }
            }

            if (_polarityId == PolarityIds.BlueOrangeBlue)
            {
                switch (significance)
                {
                    case Significance.Worse:
                        return "Lower";
                    case Significance.Same:
                        return "Same";
                    case Significance.Better:
                        return "Higher";
                }
            }
            else if (_polarityId == PolarityIds.RagHighIsGood ||
                     _polarityId == PolarityIds.RagLowIsGood)
            {
                switch (significance)
                {
                    case Significance.Worse:
                        return "Worse";
                    case Significance.Same:
                        return "Same";
                    case Significance.Better:
                        return "Better";
                }
            }

            return "Not compared";
        }
    }
}

File: IndicatorsUI/MainUI/angular-app/src/app/shared/classes/significance-formatter.ts
import { Significance } from '../../typings/FT';
import { ComparatorMethodIds, PolarityIds } from '../constants';

export class SignificanceFormatter {
    constructor(private polarityId: number, private comparatorMethodId: number) { }

    getLabel(significance: number): string {
        if (this.comparatorMethodId === ComparatorMethodIds.Quintiles) {
            switch (significance) {
                case Significance.Worse:
                    return 'Lowest quintile';
                case Significance.Same:
                    return '2nd lowest quintile';
                case Significance.Better:
                    return 'Middle quintile';
                case Significance.Worst:
                    return '2nd highest quintile';
                case Significance.Best:
                    return 'Highest quintile';
            }
        }

        if (this.polarityId === PolarityIds.BlueOrangeBlue) {
            switch (significance) {
                case Significance.Worse:
                    return 'Lower';
                case Significance.Same:
                    return 'Similar';
                case Significance.Better:
                    return 'Higher';
            }
        }

        if (this.polarityId === PolarityIds.RAGHighIsGood ||
            this.polarityId === PolarityIds.RAGLowIsGood) {

            switch (significance) {
                case Significance.Worse:
                    return 'Worse';
                case Significance.Same:
                    return 'Similar';
                case Significance.Better:
                    return 'Better';
                case Significance.Worst:
                    return 'Worst';
                case Significance.Best:
                    return 'Best';
            }
        }

        return 'Not compared';
    }

    getTargetLabel(significance: number): string {
        switch (significance) {
            case Significance.Worse:
                return 'Red';
            case Significance.Same:
                return 'Amber';
            case Significance.Better:
                return 'Green';
        }

        return 'Not compared';
    }
}
